pkg/reconciler/buildrunttlcleanup/buildrun_ttl_cleanup.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package buildrunttlcleanup
+
+import (
+	"context"
+	"time"
+
+	buildv1alpha1 "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/pkg/config"
+	"github.com/shipwright-io/build/pkg/ctxlog"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ReconcileBuildRun reconciles a BuildRun object
+
+type ReconcileBuildRun struct {
+	/* This client, initialized using mgr.Client() above, is a split client
+	   that reads objects from the cache and writes to the apiserver */
+	config *config.Config
+	client client.Client
+}
+
+func NewReconciler(c *config.Config, mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileBuildRun{
+		config: c,
+		client: mgr.GetClient(),
+	}
+}
+
+/* Reconciler makes sure the buildrun adheres to its ttl retention field and deletes it
+   once the ttl limit is hit */
+func (r *ReconcileBuildRun) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// Set the ctx to be Background, as the top-level context for incoming requests.
+	ctx, cancel := context.WithTimeout(ctx, r.config.CtxTimeOut)
+	defer cancel()
+
+	ctxlog.Debug(ctx, "Start reconciling Buildrun-ttl", namespace, request.Namespace, name, request.Name)
+
+	br := &buildv1alpha1.BuildRun{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: request.Namespace}, br)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			ctxlog.Debug(ctx, "Finish reconciling buildrun-ttl. Buildrun was not found", namespace, request.Namespace, name, request.Name)
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	condition := br.Status.GetCondition(buildv1alpha1.Succeeded)
+	if condition == nil {
```

Comment:
Just to be on the safe side.

```suggestion
	if condition == nil || condition.Status == corev1.ConditionUnknown {
```